src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -391,9 +391,13 @@ private JObject FilterRuntimes(JObject projectRoot, HashSet<string> excludedRIDs
 
         private static void WriteProject(JObject projectRoot, string projectJsonPath)
         {
-            string projectJson = JsonConvert.SerializeObject(projectRoot, Formatting.Indented);
-            Directory.CreateDirectory(Path.GetDirectoryName(projectJsonPath));
-            File.WriteAllText(projectJsonPath, projectJson + Environment.NewLine); 
+            string projectJson = JsonConvert.SerializeObject(projectRoot, Formatting.Indented) + Environment.NewLine;
+
+            if (!File.Exists(projectJsonPath) || !projectJson.Equals(File.ReadAllText(projectJsonPath)))
```

Comment:
Is doing a straight string compare of the files the right way to determine this?  I guess if you expect to be the only one writing this file it could work but it does seem like it might be a large string compare and if we are doing this hundreds of times that might cause some memory issues.